### PR TITLE
Use concord

### PR DESCRIPTION
**Pull request by backus**
Adding concord as a dependency and refactoring applicable boilerplate

### CODE REVIEW

**Comment by backus** [comment]
@dkubb @mbj this should be ready for review.  Most changes are pretty straightforward.  f76654ec56a4ab3ad4b5a384bdd7444b9e41f268 introduced two mutations which caused me to realize a way to improve the rake task description which I included in the commit before (92488ce4e22fcb23a94bd7ef89c04483a94e01b1).

File: yardstick.gemspec [hunk 1]
```diff
@@ -19,6 +19,7 @@ Gem::Specification.new do |gem|
   gem.executables      = %w[yardstick]
 
   gem.add_runtime_dependency('yard', '~> 0.8', '>= 0.8.7.2')
+  gem.add_runtime_dependency('concord', '~> 0.1')
```
Comment (mbj) [inline yardstick.gemspec]: I'd go for `~> 0.1.x` as the version as below `1.0` I use left shifted semver and there is no expected public API stability cross `0.x`. I do not plan big changes and concord should have a `1.0` already but lets keep consistence.

**Comment by mbj** [comment]
@backus I did a code only pass and only had a small flag.

**Comment by backus** [comment]
@mbj updated to read `'~> 0.1.x'`

**Comment by mbj** [comment]
@backus Code wise I'm fine, I'll leave the final merge decision to dkubb.

**Comment by dkubb** [comment]
@backus if you rebase this on top of `master` it can be merged.

**Comment by backus** [comment]
@dkubb rebased

**Pull request merged**